### PR TITLE
Setup pipelines

### PR DESCRIPTION
**Pull request by jbaxleyiii**
This PR is about setting up the infrastructure for CI/CD pipelines for this project and the CLI all together. Things setup:

## Pipelines
- [x] testing across nightlies of rust and 3 different machines
- [x] code coverage + codecov
- [x] audit of packages
- [x] build binaries and attach them to the PR
- [x] release to GitHub releases

### Testing
We could probably move to testing nightlies as a job each actual night if builds become annoyingly slow. This work does test against windows, linux, and macOS for stable and nightly builds.

### Code Coverage
Code coverage is *so so*. It does well for libraries but not great for binaries from what I can tell. I'd still like to investigate this more but it is setup to work with codecov at this point.

### Audit
This job is a nightly check and when there is an issue, it will open an issue with the found problems.

### PR binaries
This is currently done via GitHub actions which only allows people with write access to download the builds. I'd like this attached in a public way for anyone opening a PR to get a build but still have to figure that out. Possibly a place for GitHub actions?

__Before thinning__
image

__After thinning__
image

### GitHub Releases
This is done with GitHub actions and done via a tag that starts with `v`. I'll document out the current release steps once I'm happy with it

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+To ensure you are ready, you should be able to run this:
+
+`cargo --version` 
+
+and get something back that looks like this:
+
+`cargo 1.42.0 (86334295e 2020-01-31)`
+
+**Testing the project**
+Rust has great built-in test tooling. You can write tests in-line or under the tests folder for each crate. Tests under the `tests` folder should be integration whereas inline should be unit tests (as I understand from the conventions). You can run `cargo test` to test all of the crates at once.
```
Comment (JakeDawkins) [inline Contributing.md:114]:
```suggestion
Rust has great built-in test tooling. You can write tests in-line or under the tests folder for each crate. Tests under the `tests` folder should be integration whereas inline should be unit tests. You can run `cargo test` to test all of the crates at once. For more info on how to write tests, look to similar parts of the codebase or read the [rustlang.org article](https://doc.rust-lang.org/book/ch11-00-testing.html) on testing
```

File: Contributing.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+
+**Opening a Pull Request**
+Once you have fixed a bug, created the next great feature, or improved the project in some other way it is time to open a pull request! To learn more about how to do this, follow [this great guide](https://opensource.guide/how-to-contribute/#opening-a-pull-request). When you open a PR, the pipeline steps will run to ensure the changes will work for users of this library. To learn more about the pipeline, keep reading below!
+
+## Pipelines
+
+This project uses GitHub Actions to run a continuous integration and delivery pipeline. Every code change will be run against a few steps to help keep the project running at its peak ability
+
+* **CLA Check**: If you haven’t signed the Apollo CLA, a bot will comment on your PR asking you to do this
```
Comment (JakeDawkins) [inline Contributing.md:127]: Should we link to the that here? Small improvement that could definitely be skipped

File: Contributing.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+* **Tests**: The CI will run the `cargo test` command across there different architectures (Mac OS, Linux, and Windows). If your build fails on a platform that can’t test on don’t worry! The team will be able to help you out as we run all three platforms to make sure everyone has a great experience.
+* **Coverage**: This project runs a job to collect information on how much of the library has been tested using code coverage tools. This is a WIP but you may see a status check related to how you have improved (or lowered 😢) the amount covered. Don’t worry, it happens to all of us and we are hear to help out!
```
Comment (JakeDawkins) [inline Contributing.md:129]:
```suggestion
* **Coverage**: This project runs a job to collect information on how much of the library has been tested using code coverage tools. This is a WIP but you may see a status check related to how you have improved (or lowered 😢) the amount covered. Don’t worry, it happens to all of us and we are here to help out!
```

File: Readme.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+<img src="https://raw.githubusercontent.com/apollographql/space-kit/master/src/illustrations/svgs/rocket1.svg" width="100%" height="144">
+
+# Apollo CLI
+
+> This is a new version of the Apollo CLI. If you are looking for the currently released and supported version, please go to the [docs page](https://www.apollographql.com/docs/devtools/cli/) or the [Apollo Tooling repository](https://github.com/apollographql/apollo-tooling). As this library matures, we will being pointing users to use the new CLI for certain features in a gradual role out.
```
Comment (JakeDawkins) [inline Readme.md:5]:
```suggestion
> This is a new version of the Apollo CLI. If you are looking for the currently released and supported version, please go to the [docs page](https://www.apollographql.com/docs/devtools/cli/) or the [Apollo Tooling repository](https://github.com/apollographql/apollo-tooling). As this library matures, we will being pointing users to use the new CLI for certain features in a gradual roll out.
```

File: Contributing.md (hunk 2 above)
Comment (jbaxleyiii) [reply Contributing.md:127]: I still need to turn it on haha

File: Contributing.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+### Small bug fixes
+
+For a small bug fix change (less than ~20 lines of code changed), feel free to open a pull request. We’ll try to merge it as fast as possible and ideally publish a new release on the same day. The only requirement is, make sure you also add a test that verifies the bug you are trying to fix.
+
+### Suggesting features
+
+Most of the features in Apollo Client came from suggestions by you, the community! We welcome any ideas about how to make Apollo  better for your use case. Open up a new feature request / discussion issue with your details.
+
+**Note**: Feature requests and non-bug related discussions are no longer managed in this repo's issue tracker. Feature request and/or discussions opened here will be closed.
```
Comment (JakeDawkins) [inline Contributing.md:56]: If we drop the mention of the feature request repo, this should go too :)

File: Contributing.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+## Big Changes or New Features
+
+For significant changes to a repository, it’s important to settle on a design before starting on the implementation. This way, we can make sure that major improvements get the care and attention they deserve. Since big changes can be risky and might not always get merged, it’s good to reduce the amount of possible wasted effort by agreeing on an implementation design/plan first.
+
+
+1. **Open an issue.** Open an issue about your bug in this repo, or about your feature request
```
Comment (JakeDawkins) [inline Contributing.md:63]:
```suggestion
1. **Open an issue.** Open an issue about your bug or feature request in this repo. 
```

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -12,11 +12,11 @@ jobs:
         build:
           [
             linux-stable,
-            linux-nightly,
+            # linux-nightly,
```
Comment (queerviolet) [inline .github/workflows/test.yml:15]: is testing on the nightlies a pain?

**Pull request merged**